brave/browser-laptop
# Making the Clear Browsing Data panel stateful

Fixes #4082 
Majorly, the change affects 2 places in the UI. The Clear Browsing
Data… option in the History menu and the Clear Browsing Data Now…
button in the Security tab in Preferences. We store the state of the
toggled options when the user clears anytime, so that the next time
he/she opens this dialog, the options are pre-populated as per the last
selected options

From a technical perspective, we have moved away from saving the
current state of the SwitchControls in the windowState. We only save
the visibility on the windowState. The whole state is managed by the
panel itself and just updates the appState on clicking Clear.

Auditors:
@bsclifton 

## Test plan
1. Launch Brave visit preferences
2. On the left, choose Security tab
3. Click `Clear Browsing Data`
4. Choose very specific options; like `download history` and `all site cookies` and click `Clear`
5. After window closes, click `Clear Browsing Data` again 
6. The same options should be selected now. Close the window with `Cancel`.
7. Pick `Clear Browsing Data` from the `History` menu
8. Confirm the same options from step 4 are shown. Hit cancel to close
9. Load about:history and click `Clear Browsing Data`
10. Confirm the same options from step 4 are shown. Hit cancel to close

## Automated test plan
```bash
#window 1
npm run watch
#window 2
npm run uitest -- --grep="state is preserved"
npm run uitest -- --grep="Clear Browsing Panel"
npm run uitest -- --grep="urlBar tests"
```
